### ISSUE TITLE
PLE1142 should flag await in lambda

### ISSUE THREAD

**Issue by jakkdl** [astral-sh/ruff#15783]
### Description

```python
"""silence D100"""
async def afoo() -> None:
    """silence D103"""
    return lambda: await afoo()
```
```
$ ruff check --select=ALL foo.py
[...]
All checks passed!
```
```
$ python foo.py
  File "/tmp/ruff/foo.py", line 6
    return lambda: await afoo()
                   ^^^^^^^^^^^^
SyntaxError: 'await' outside async function
```
```
$ ruff --version
ruff 0.9.3
```

**Comment by jakkdl** [comment]
I also reported a similar issue a few months back with #14167

**Comment by MichaReiser** [comment]
Thanks @InSyncWithFoo. Yes, we want to tackle this as part of #11934
